source/MaterialXContrib/MaterialXNode/MaterialXShadingNodeImpl.cpp

```diff
@@ -79,27 +81,58 @@ MStatus bindFileTexture(MHWRender::MShaderInstance& shader,
+            if (udimTiles.size() != udimPaths.size() * 2)
             {
-                VP2TextureUniquePtr texture(textureManager->acquireTexture(imagePath.asString().c_str(), MaterialX::EMPTY_STRING.c_str()));
-                if (texture)
-                {
-                    MHWRender::MTextureAssignment textureAssignment;
-                    textureAssignment.texture = texture.get();
-                    status = shader.setParameter(parameterName.c_str(), textureAssignment);
-
-                    // Get back the texture description
-                    texture->textureDescription(textureDescription);
-                }
+                throw mx::Exception("Failed to resolve UDIM information properly for file:" + fileName);
             }
+            MStringArray mTilePaths;
+            MFloatArray mTilePositions;
+            MColor undefinedColor;
+            MStringArray failedTilePaths;
+            MFloatArray uvScaleOffset;
+            for (size_t i=0; i<udimPaths.size(); i++)
+            {
+                MaterialX::FilePath resolvedPath = MaterialXMaya::findInSubdirectories(searchPath, udimPaths[i]);
+                const mx::FilePath& udimPath = resolvedPath;
+                mTilePaths.append(udimPath.asString().c_str());
+                mTilePositions.append(static_cast<float>(udimTiles[i * 2]));
+                mTilePositions.append(static_cast<float>(udimTiles[i * 2 + 1]));
+            }
+            const unsigned int UDIM_BAKE_WIDTH = 4096;
+            const unsigned int UDIM_BAKE_HEIGHT = 4096;
+            // Note: we do not use the uv scale and offset. Ideally this should be used for the texture lookup code
+            // but at this point the shader code has already been generated.
+            texture = textureManager->acquireTiledTexture(fileName.c_str(), mTilePaths, mTilePositions, 
+                                                          undefinedColor, UDIM_BAKE_WIDTH, UDIM_BAKE_HEIGHT, 
+                                                          failedTilePaths, uvScaleOffset);
         }
         else
         {
-            // TODO: Add a fallback image here to use.
-            std::cerr << "Cannot find image file: " << fileName << ". Search paths: "
-                      << searchPath.asString() << std::endl;
+            MaterialX::FilePath imagePath = MaterialXMaya::findInSubdirectories(searchPath, fileName);
+            if (!imagePath.isEmpty())
+            {
+                texture = textureManager->acquireTexture(imagePath.asString().c_str(), MaterialX::EMPTY_STRING.c_str());
+            }
+        }
+
+        MHWRender::MTextureAssignment textureAssignment;
+        VP2TextureUniquePtr texturePtr(texture);
+        textureAssignment.texture = texture;
+        if (texture)
+        {
+            // Get back the texture description
+            texture->textureDescription(textureDescription);
```

Comment:
@ppenenko . As discussed we always bind even if the texture is null. The shader code will detect this and use the default color so we don't require a fallback texture here.